### PR TITLE
Feature/pa 55 add logout

### PR DESCRIPTION
**Pull request by TondeMoon**
pa-55

add logout option
add test in user Reducer test

## How it can be tested?
test in user Reducer test

## PR Checklist

Please check if your PR fulfills the following requirements:
 - [X] I tested my code

### CODE REVIEW

File: frontend/src/components/personal/PreferencesPage.tsx [hunk 1]
```diff
@@ -39,7 +40,7 @@ export default function PreferencesPage (): React.ReactElement {
     setEmail(e.target.value)
   }, [setEmail])
 
-  const [id, setId] = useState(1)
+  const [id, setId] = useState(user? user.id : 1)
```
Comment (ivan-gerasin) [inline frontend/src/components/personal/PreferencesPage.tsx]: I think we should not use default value here

File: frontend/src/components/personal/PreferencesPage.tsx [hunk 2]
```diff
@@ -50,6 +51,11 @@ export default function PreferencesPage (): React.ReactElement {
     dispatch(changeMyData({nickname, email, id}))
   }, [dispatch, nickname, email, id])
 
+  const logOut = useCallback((e) => {
```
Comment (ivan-gerasin) [inline frontend/src/components/personal/PreferencesPage.tsx:50]: Why does logout is on preference page? Can it be on every page?

Comment (TondeMoon) [reply frontend/src/components/personal/PreferencesPage.tsx:50]: add logout to profile page.

File: frontend/src/components/personal/PreferencesPage.tsx (hunk 1 above)
Comment (TondeMoon) [reply frontend/src/components/personal/PreferencesPage.tsx]: corrected as required

File: frontend/src/components/personal/PreferencesPage.tsx (hunk 2 above)
Comment (ivan-gerasin) [reply frontend/src/components/personal/PreferencesPage.tsx:50]: I thought about navbar on each page :)

**Review by ivan-gerasin — APPROVED** [review]

**Pull request merged**